### PR TITLE
Added function pointer comparison to FunctionPointerBase.

### PR DESCRIPTION
**Pull request by marcuschangarm**
Take 2.

### CODE REVIEW

File: mbed-util/FunctionPointerBase.h [hunk 1]
```diff
@@ -31,6 +31,14 @@ class FunctionPointerBase {
         return (_membercaller != NULL) && (_object != NULL);
     }
 
+    bool operator==(const FunctionPointerBase& other) {
```
Comment (rgrover) [inline mbed-util/FunctionPointerBase.h]: this can be a const method. i.e. `bool operator== (const FunctionPointerBase & other) const {`

**Comment by marcuschangarm** [comment]
@rgrover Thank you, const now added.

**Comment by 0xc0170** [comment]
@bremoran

File: mbed-util/FunctionPointerBase.h [hunk 2]
```diff
@@ -38,6 +46,7 @@ class FunctionPointerBase {
     virtual void clear() {
         _membercaller = NULL;
         _object = NULL;
+        memset(_member, 0, sizeof(_member));
```
Comment (bremoran) [inline mbed-util/FunctionPointerBase.h]: Given that _member is guaranteed to be 4-byte aligned, is memset the best choice here?

**Comment by bremoran** [comment]
Other than alignment question, looks good.

File: mbed-util/FunctionPointerBase.h (hunk 2 above)
Comment (marcuschangarm) [inline mbed-util/FunctionPointerBase.h]: I used memset to make the code easy to read.

Comment (bremoran) [inline mbed-util/FunctionPointerBase.h]: A private function that sets the member to 0 would be just as readable and more efficient.

**Comment by 0xc0170** [comment]
Please rebase

**Comment by marcuschangarm** [comment]
Continues in https://github.com/ARMmbed/core-util/pull/29

**Pull request closed without merging**